src/Views/HardwareView.vala

```diff
@@ -488,12 +479,41 @@ public class About.HardwareView : Gtk.Grid {
         string regex;
         string replacement;
     }
+
+    private void get_system_interface_instance () {
+        if (system_interface == null) {
+            try {
+                system_interface = Bus.get_proxy_sync (
+                    BusType.SYSTEM,
+                    "org.freedesktop.hostname1",
+                    "/org/freedesktop/hostname1"
+                );
+            } catch (GLib.Error e) {
+                warning ("%s", e.message);
+            }
+        }
+    }
+
+    private string get_host_name () {
+        get_system_interface_instance ();
+
+        string hostname = system_interface.pretty_hostname;
```

Comment:
Same here with the null check and we could fall back to Environment.get_host_name () in case `system_interface` is null